app/services/vote_service.rb

```diff
@@ -1,10 +1,13 @@
 module VoteService
-  def self.user_vote(post, option)
+  def self.user_vote(post)
     begin
       VotePost.transaction do
         post.save!
-        option.count += 1
-        option.save!
+        options = VoteOption.find(post.vote_option_ids)
+        options.each do |o|
```

Comment:
> How is the amount of votes limited to the choices?

We set `choices` when we create/edit a `Vote`. When we run `post.save!` the validation `:option_details` in `VotePost` checks that `vote_option_ids.count <= vote.choices` and that no duplicates are included :)